src/Raven.Server/Documents/Handlers/Processors/Streaming/StreamingHandlerProcessorForGetDocs.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Server.Documents.Handlers.Streaming;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Utils;
+using Raven.Server.Utils.Enumerators;
+using Raven.Server.Json;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers.Processors.Streaming
+{
+    internal class StreamingHandlerProcessorForGetDocs : AbstractStreamingHandlerProcessorForGetDocs<DatabaseRequestHandler, DocumentsOperationContext>
+    {
+        public StreamingHandlerProcessorForGetDocs([NotNull] DatabaseRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+
+        protected override async ValueTask GetDocumentsAndWriteAsync(DocumentsOperationContext context, int start, int pageSize, string startsWith,
+            string excludes, string matches, string startAfter)
+        {
+            using (context.OpenReadTransaction())
+            {
+                var initialState =
+                    new DocsStreamingIterationState(context, RequestHandler.Database.Configuration.Databases.PulseReadTransactionLimit)
+                    {
+                        Start = start, 
+                        Take = pageSize
+                    };
+
+                if (HttpContext.Request.Query.ContainsKey("startsWith"))
```

Comment:
Let's use dedicated request handler methods to extract query string parameters. You probably can check if 'startsWith' is not null or empty?

Reply:
Checking if null because empty string is allowed